api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/SchedulerHeaderInterceptor.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import static uk.gov.dhsc.htbhf.requestcontext.RequestIdFilter.REQUEST_ID_HEADER;
+import static uk.gov.dhsc.htbhf.requestcontext.RequestIdFilter.SESSION_ID_HEADER;
+
+/**
+ * Creates an http interceptor used to be used by the scheduler. The interceptor
+ * generates a request and session id and appends them to the incoming request headers.
+ * This is required as {@link uk.gov.dhsc.htbhf.requestcontext.HeaderInterceptor} only works in
+ * the scope of a web request.
```

Comment:
Not quite true, the `HeaderInterceptor` works with a `RequestContext` - which we need to be updating when we create the context for a scheduled job (so that log messages identify which scheduled job they are running). So the session id needs to be the id of the current job. We do want the request id to be different every time we make a request, I expect.